packages/main/src/plugin/kubernetes-client.ts

```diff
@@ -1311,4 +1319,304 @@ export class KubernetesClient {
+    if (!isJobDeleted) {
+      throw new Error(`job "${name}" in namespace "${namespace}" was not deleted within the expected timeframe`);
+    }
+
+    const labelSelector = `job-name=${name}`;
+    const isPodsDeleted = await this.waitForPodsDeletion(coreApi, namespace, labelSelector);
+    if (!isPodsDeleted) {
+      throw new Error(
+        `not all pods with selector "${labelSelector}" in namespace "${namespace}" were deleted within the expected timeframe`,
+      );
+    }
+
+    delete existingJob.metadata!.creationTimestamp;
+    delete existingJob.metadata!.resourceVersion;
+    delete existingJob.metadata!.selfLink;
+    delete existingJob.metadata!.uid;
+    delete existingJob.metadata!.ownerReferences;
+    delete existingJob.status;
+    delete existingJob.spec!.selector;
+    if (existingJob.spec!.template.metadata!.labels) {
+      delete existingJob.spec!.template.metadata!.labels['controller-uid'];
+      delete existingJob.spec!.template.metadata!.labels['batch.kubernetes.io/controller-uid'];
+      delete existingJob.spec!.template.metadata!.labels['batch.kubernetes.io/job-name'];
+      delete existingJob.spec!.template.metadata!.labels['job-name'];
+    }
+    if (existingJob.metadata?.labels) {
+      delete existingJob.metadata.labels['controller-uid'];
+      delete existingJob.metadata.labels['batch.kubernetes.io/controller-uid'];
+      delete existingJob.metadata.labels['batch.kubernetes.io/job-name'];
+      delete existingJob.metadata.labels['job-name'];
+    }
+
+    await batchApi.createNamespacedJob(namespace, existingJob);
+  }
+
+  protected async waitForJobDeletion(
+    batchApi: BatchV1Api,
+    name: string,
+    namespace: string,
+    timeout: number = 60000,
+  ): Promise<boolean> {
+    const startTime = Date.now();
+
+    while (Date.now() - startTime < timeout) {
+      try {
+        await batchApi.readNamespacedJobStatus(name, namespace);
+        await new Promise(resolve => setTimeout(resolve, 1000));
+      } catch (e) {
+        const error = e ?? {};
+        if (typeof error === 'object' && 'response' in error) {
+          const axiosError = error as { response: { statusCode: number } };
+          if (axiosError.response.statusCode === 404) {
+            return true;
+          } else {
+            throw e;
+          }
+        } else {
+          throw e;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  protected async waitForPodsDeletion(
+    coreApi: CoreV1Api,
+    namespace: string,
+    selector: string,
+    timeout: number = 60000,
+  ): Promise<boolean> {
+    const startTime = Date.now();
+
+    while (Date.now() - startTime < timeout) {
+      const { body: podList } = await coreApi.listNamespacedPod(
+        namespace,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        selector,
+      );
+      if (podList.items.length === 0) {
+        return true;
+      }
```

Comment:
Should be using watch with resourceVersion, to avoid use case when pod deleted before while loop started (see https://kubernetes.io/docs/reference/using-api/api-concepts/#efficient-detection-of-changes).

Reply:
if the pod deleted before while loop started, the method would just return true, indicating, that pod had been already deleted. should we refactor this method to use watch api?

Reply:
Works for me as it is, watch would be cleaner, but it is optional.